### ISSUE TITLE
Program killed accidentally

### ISSUE THREAD

**Issue by o0t1ng0o** [adepierre/Caffe_CycleGAN#4]
Hi, @adepierre ~
I found a weird problem when training.
It is weird that the program is killed unexpectedly when iteration is 62250.
But nobody kills the program.
There are 753 image pairs for training.

Loss on real A data: 0.0568343 on generated A data: 0.0570172
Loss on real B data: 0.0519266 on generated B data: 0.0524207
Cyclic loss on A data: 0.229246
Cyclic loss on B data: 0.164716
I0712 23:00:19.389642 45804 sgd_solver.cpp:112] Iteration 62250, lr = 0.0002
I0712 23:00:19.398865 45804 sgd_solver.cpp:112] Iteration 62250, lr = 0.0002
I0712 23:00:19.401080 45804 sgd_solver.cpp:112] Iteration 62250, lr = 0.0002
I0712 23:00:19.401469 45804 sgd_solver.cpp:112] Iteration 62250, lr = 0.0002
Killed

Could you please answer this question?
Many thanks!!

**Comment by adepierre** [comment]
Hi

I'm not sure of what you mean by "killed". Isn't it just the end of the training?

You should see some "A_to_B_epoch_N.png" and "A_to_B.png" images from the training in the directory. Are they here?

If you want to restart the training from a checkpoint you can set the --snapshot_gen_X_to_Y flags with Caffe's snapshots files corresponding to the checkpoint you want to restart from (something like CycleGAN_Discriminator_X_to_Y.solverstate)

**Comment by o0t1ng0o** [comment]
I am pretty sure that it is not the end of my training.
'Caurse i set the start_epoch =0, end_epoch = 200.
There are "A_to_B_epoch_42.png" and "B_to_A_epoch_42.png" images in the directory.  
I think that it should run 753 X 200 = 150600 iterations. (there are 753 images in training set A.)
But it terminated when iteration is 62250 = 753 X 82 + 500.

Is there any problem in the code(main.cpp)?

**Comment by o0t1ng0o** [comment]
I found that it may be a problem about runing out of swap.
[https://stackoverflow.com/questions/726690/what-killed-my-process-and-why?answertab=active#tab-top](url)
Maybe I need to extend the space of swap or enable the application to consume less memory...

**Comment by adepierre** [comment]
You're right about the number of iterations, it's min(num_images_A, num_images_B) * num_epoch / batch_size.

I am not sure I can reproduce this bug now, but I'm pretty confident the program was not killed because of bug in the code. Maybe it's a memory issue or something like that. If you use the solverstate files you should be able to start the training from approximately where it stopped
